go_utils/photo_download.py

```diff
@@ -284,17 +361,42 @@ def get_lc_download_targets(
     """
     arguments = locals()
     targets = set()
+    num_invalid_photos = 0
 
     def get_photo_args(url, latitude, longitude, direction, date, lc_id):
-        if pd.isna(url) or "https" not in url:
-            return
-
-        date_str = pd.to_datetime(str(date)).strftime("%Y-%m-%d")
-        photo_id = get_globe_photo_id(url)
+        if not pd.isna(url) and "https" in url:
+            date_str = pd.to_datetime(str(date)).strftime("%Y-%m-%d")
+            photo_id = get_globe_photo_id(url)
 
-        name = f"lc_{direction}_{round(latitude, 5)}_{round(longitude, 5)}_{date_str}_{lc_id}_{photo_id}.png"
-        name = remove_bad_characters(name)
-        targets.add((url, directory, name))
+            # Constructs Photo Name using given included fields and additional information
+            def build_name():
+                name = "lc_"
+                if additional_name_stem and additional_name_stem != "":
+                    name += f"{additional_name_stem}_"
+
+                if include_in_name:
+                    for field in list(include_in_name):
+                        if field in list(name_fields):
+                            name += f"{name_fields[field]}_"
+                name += f"{photo_id}.png"
+                name = remove_bad_characters(name)
+                return name
+
+            name_fields = {
+                "direction": direction,
+                "latitude": round(latitude, 5),
+                "longitude": round(longitude, 5),
+                "date_str": date_str,
+                "lc_id": lc_id,
+            }
+
+            if not pd.isna(photo_id) and int(photo_id) >= 0:
+                name = build_name()
+                targets.add((url, directory, name))
+            else:
+                num_invalid_photos + 1
+        else:
+            num_invalid_photos + 1
```

Comment:
Not sure if this increments num_invalid_photos. It might also be good to add [tests ](https://docs.pytest.org/en/6.2.x/warnings.html#warns) to make sure this works
```suggestion
                num_invalid_photos += 1
        else:
            num_invalid_photos += 1
```

Reply:
added tests! -- Python doesn't allow modifications to an outer variable from within a nested function, so I found a workaround by creating a 1-element dict of the `num_invalid_photos`